src/language-js/print/assignment.js

```diff
@@ -243,6 +247,31 @@ function isAssignmentOrVariableDeclarator(node) {
   return isAssignment(node) || node.type === "VariableDeclarator";
 }
 
+function isCompletxTypeAliasParams(node) {
+  const typeParams = getTypeParametersFromTypeAlias(node);
+  if (isNonEmptyArray(typeParams)) {
+    if (
+      typeParams.length > 1 &&
+      !!getLast(typeParams).default &&
+      typeParams.every((param) => !!param.constraint)
+    ) {
```

Comment:
Right now we only have the condition to fix just #10857, but I think we can improve here based on feedback in the future.